deploy/helm/kubecf/assets/scripts/jobs/cf-cli-6-linux/setup_internal_security_group.sh

```diff
@@ -3,60 +3,218 @@
+# Write the CF_API_CA_CERT and UAA_CA_CERT environment variable contents to PEM files.
 if [ "${CF_API_CA_CERT:-}" != "" ]; then
     cf_api_ca_cert_pem="${DATA_DIR}/cf_api_ca_cert.pem"
     echo -n "${CF_API_CA_CERT}" > "${cf_api_ca_cert_pem}"
-    export SSL_CERT_FILE="${cf_api_ca_cert_pem}"
+    uaa_ca_cert_pem="${DATA_DIR}/uaa_ca_cert.pem"
+    echo -n "${UAA_CA_CERT}" > "${uaa_ca_cert_pem}"
 fi
 
+# wait_for_endpoint waits for an endpoint to be ready. It fails if the error is related to the
+# certificate.
+function wait_for_endpoint() {
+    local ca_cert
+    ca_cert="$1"
+    local endpoint
+    endpoint="$2"
+    while true; do
+        set +o errexit
+        output=$(curl --cacert "${ca_cert}" --fail --head --silent --show-error "${endpoint}" 2>&1)
+        code=$?
+        set -o errexit
+        if [ "${code}" == "60" ]; then
+            >&2 echo "${output}"
+            return 1
+        fi
+        if [ "${code}" == "0" ]; then break; fi
+        sleep 1
+    done
+}
+
 # Wait for the CF API to be accessible.
 echo "Waiting for the API to be accessible..."
-while true; do
-    set +o errexit
-    output=$(curl --fail --head "${CF_API}/v2/info" 2>&1)
-    code=$?
-    set -o errexit
-    if [ "${code}" == "60" ]; then
-        >&2 echo "${output}"
-        exit 1
-    fi
-    if [ "${code}" == "0" ]; then break; fi
-    sleep 1
-done
-
-# Setup the cf-cli.
-cf api "${CF_API}"
-cf auth --client-credentials
-
-# Define the security group JSON.
+wait_for_endpoint "${cf_api_ca_cert_pem}" "${CF_API}/v2/info"
+
+# Wait for UAA to be accessible.
+echo "Waiting for UAA to be accessible..."
+wait_for_endpoint "${uaa_ca_cert_pem}" "${UAA_URL}/info"
+
+# Construct the security group JSON definition.
+sec_group_name="internal-${POD_NAME}"
 sec_group_json=$(cat <<EOF
-[{
-    "protocol": "tcp",
-    "description": "Allow traffic to ${POD_NAME}",
-    "destination": "${POD_IP}/32",
-    "ports": "${PORTS}"
-}]
+{
+    "name": "${sec_group_name}",
+    "rules": [{
+        "protocol": "tcp",
+        "description": "Allow traffic to ${POD_NAME}",
+        "destination": "${POD_IP}/32",
+        "ports": "${PORTS}"
+    }]
+}
 EOF
 )
 
-# Create the security group if it doens't exist, otherwise update it.
-sec_group_name="internal-${POD_NAME}"
-if ! cf security-group "${sec_group_name}"; then
-    cf create-security-group "${sec_group_name}" <(echo -n "${sec_group_json}")
-    cf bind-staging-security-group "${sec_group_name}"
-    cf bind-running-security-group "${sec_group_name}"
+# get_access_token returns a token to be used on the CF API calls.
+function get_access_token() {
+    scope="scope=cloud_controller.admin"
+    grant_type="grant_type=client_credentials"
+    client_id="client_id=${CF_USERNAME}"
+    client_secret="client_secret=${CF_PASSWORD}"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${uaa_ca_cert_pem}" \
+        --request POST \
+        --data "${scope}&${grant_type}&${client_id}&${client_secret}" \
```

Comment:
```suggestion
        --data "${scope}&${grant_type}&${client_id}&${client_secret}" \
        --data scope="${scope}" \
        --data grant_type="${grant_type}" \
        --data client_id="${CF_USERNAME}" \
        --data client_secret="${CF_PASSWORD}" \
```
(Of course with appropriate changes to the values)

Reply:
I'll accept this change, but won't do via Github since this will break if not changing the values.

Reply:
Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/485/commits/da0e3bd19f83bf58ffaa06ff6972d0ed46930e8e.